cpp/include/cudf/strings/char_types/char_types.hpp

```diff
@@ -170,6 +170,21 @@ std::unique_ptr<column> is_integer(
   strings_column_view const& strings,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns a boolean column identifying strings in which all
+ * characters are valid for conversion to integers.
+ * 
+ * @param strings Strings instance for this operation.
+ * @param allow_decimal identification whether the format is decimal or not
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New column of boolean results for each string.
+ */
+std::unique_ptr<column> is_valid_integer(
+  strings_column_view const& strings,
+  bool allow_decimal,
```

Comment:
This parameter doesn't seem like it belongs. Fixed-point decimal numbers aren't integers. This seems like it should be `is_valid_fixed_point` or something similar.

Reply:
I agree. I've been looking up `is_numeric(), is_decimal(), etc` type APIs in other languages and cannot find one that matches the requirements for #7080 .  Perhaps `is_valid_numeric()` with _allow-decimal_  and _allow-exponent_ flags. Perhaps this is too Spark specific.

Reply:
The name does not matter too much to me.  `is_valid_numeric` is OK, but I think I like `is_valid_fixed_point` best because it does appear to match with a fixed point number.

Reply:
Instead of all of these type specific APIs, couldn't we just have:
```
unique_ptr<column> is_valid_element(strings_column_view strings, data_type type);
```
i.e., verifies that the string elements in `strings` can be parsed as the specified `type`?

Reply:
`is_valid_element` is fine with me. The main issue would be decimal vs not decimal for an integer.  I think we could address that with some flags passed to `is_valid_element`. I also assume that this PR would only be for integer types and other types would be added later?

Reply:
> The main issue would be decimal vs not decimal for an integer

I don't understand the problem. This is what I was thinking:

If you want to know if `"12345"` is a valid integer you'd do:
```
is_valid_element( "12345", data_type{INT32})`
```

If you want to know if it's a valid fixed-point decimal, you do:
```
is_valid_element("12345", data_type{decimal64, scale, etc.})
```

Reply:
And yes, `is_valid_element` can be a different PR.